### PR TITLE
(fix): gpu ci uses max version from `pyproject.toml`

### PR DESCRIPTION
**Pull request by ilan-gold**
- [ ] Closes #
- [ ] Tests added
- [ ] Release note added (or unnecessary)

### CODE REVIEW

File: .github/workflows/test-gpu.yml [hunk 1]
```diff
@@ -51,10 +51,17 @@ jobs:
       - name: Nvidia SMI sanity check
         run: nvidia-smi
 
+      - name: Extract max Python version from classifiers
+        run: |
+          sudo apt-get update && sudo apt-get install -y cargo && cargo install toml-cli
```
Comment (flying-sheep) [inline .github/workflows/test-gpu.yml:56]: You could also use this: https://github.com/marketplace/actions/cargo-install

Comment (ilan-gold) [reply .github/workflows/test-gpu.yml:56]: ah was looking for something like this just now.  I also thought cargo came on the ubuntu images by default but evidently not `install`? 

Also, do you think we should be doing this for all of our CI i.e., extracting from `pyproject.toml`?

Comment (ilan-gold) [reply .github/workflows/test-gpu.yml]: Ok I think that is only for rust projects: https://github.com/scverse/anndata/actions/runs/11330444061/job/31508293550?pr=1714

Comment (flying-sheep) [reply .github/workflows/test-gpu.yml:56]: I still think that using that action is better, because it sets up PATH and uses caching.

Comment (ilan-gold) [reply .github/workflows/test-gpu.yml:56]: https://github.com/scverse/anndata/actions/runs/11330444061/job/31508293550 Not sure

Comment (ilan-gold) [reply .github/workflows/test-gpu.yml:56]: I read this as `cargo` not being on the path

**Review by flying-sheep — APPROVED** [review]
Awesome! I’m not a huge fan of almost all[^1] short-form CLI parameters, so if you could switch to long-form I’d even be happier.

Also `\K` means something regarding capture right? If I’m right, maybe use a capture group instead, I think people are more likely to know that one. Otherwise please add a comment explaining what it is.

[^1]: except super-well known ones like `something -vv` or `grep -eP`

**Review by flying-sheep — APPROVED** [review]
Ah great, now I understand what’s going on!

**Pull request merged**